source/module_cell/read_pp_upf201.cpp

```diff
@@ -132,28 +132,52 @@ int Pseudopot_upf::read_pseudo_upf201(std::ifstream &ifs)
 
 	
 
-	ModuleBase::GlobalFunc::SCAN_BEGIN(ifs, "<PP_R");
-	ModuleBase::GlobalFunc::READ_VALUE(ifs, word); // type size columns
-	delete[] r;
-	delete[] rab;
-	assert(mesh>0);
-	this->r = new double[mesh];
-	this->rab = new double[mesh];
-	ModuleBase::GlobalFunc::ZEROS(r,mesh);
-	ModuleBase::GlobalFunc::ZEROS(rab,mesh);
-	for (int ir = 0;ir < mesh;ir++)
+	if (ModuleBase::GlobalFunc::SCAN_BEGIN(ifs, "<PP_R"))
+	{
+		ModuleBase::GlobalFunc::READ_VALUE(ifs, word); // type size columns
+		delete[] r;
+		delete[] rab;
+		assert(mesh>0);
+		this->r = new double[mesh];
+		this->rab = new double[mesh];
+		ModuleBase::GlobalFunc::ZEROS(r,mesh);
+		ModuleBase::GlobalFunc::ZEROS(rab,mesh);
+		for (int ir = 0;ir < mesh;ir++)
+		{
+			ifs >> this->r[ir];
+		}
+	}
+	else if (ModuleBase::GlobalFunc::SCAN_BEGIN(ifs, "<PP_R>"))
```

Comment:
Please don't repeat the code.
You can use inline function or other method.

Reply:
Inline function can't contain for-loop, so I define four auxiliary functions to aviod repeating the code.